bridge/bridge.go

```diff
@@ -273,86 +275,102 @@ func (b *Bridge) newService(port ServicePort, isgroup bool) *Service {
+		}
+		service.Port = p
+
+		if b.config.UseIpFromLabel != "" {
+			containerIp := container.Config.Labels[b.config.UseIpFromLabel]
+			if containerIp != "" {
+				slashIndex := strings.LastIndex(containerIp, "/")
+				if slashIndex > -1 {
+					service.IP = containerIp[:slashIndex]
+				} else {
+					service.IP = containerIp
+				}
+				log.Println("using container IP " + service.IP + " from label '" +
+					b.config.UseIpFromLabel  + "'")
 			} else {
-				service.IP = containerIp
+				log.Println("Label '" + b.config.UseIpFromLabel +
+					"' not found in container configuration")
 			}
-			log.Println("using container IP " + service.IP + " from label '" +
-				b.config.UseIpFromLabel  + "'")
-		} else {
-			log.Println("Label '" + b.config.UseIpFromLabel +
-				"' not found in container configuration")
 		}
-	}
 
-	// NetworkMode can point to another container (kuberenetes pods)
-	networkMode := container.HostConfig.NetworkMode
-	if networkMode != "" {
-		if strings.HasPrefix(networkMode, "container:") {
-			networkContainerId := strings.Split(networkMode, ":")[1]
-			log.Println(service.Name + ": detected container NetworkMode, linked to: " + networkContainerId[:12])
-			networkContainer, err := b.docker.InspectContainer(networkContainerId)
-			if err != nil {
-				log.Println("unable to inspect network container:", networkContainerId[:12], err)
-			} else {
-				service.IP = networkContainer.NetworkSettings.IPAddress
-				log.Println(service.Name + ": using network container IP " + service.IP)
+		// NetworkMode can point to another container (kuberenetes pods)
+		networkMode := container.HostConfig.NetworkMode
+		if networkMode != "" {
+			if strings.HasPrefix(networkMode, "container:") {
+				networkContainerId := strings.Split(networkMode, ":")[1]
+				log.Println(service.Name + ": detected container NetworkMode, linked to: " + networkContainerId[:12])
+				networkContainer, err := b.docker.InspectContainer(networkContainerId)
+				if err != nil {
+					log.Println("unable to inspect network container:", networkContainerId[:12], err)
+				} else {
+					service.IP = networkContainer.NetworkSettings.IPAddress
+					log.Println(service.Name + ": using network container IP " + service.IP)
+				}
 			}
 		}
-	}
 
-	if port.PortType == "udp" {
-		service.Tags = combineTags(
-			mapDefault(metadata, "tags", ""), b.config.ForceTags, "udp")
-		service.ID = service.ID + ":udp"
-	} else {
-		service.Tags = combineTags(
-			mapDefault(metadata, "tags", ""), b.config.ForceTags)
-	}
+		if port.PortType == "udp" {
+			service.Tags = combineTags(
+				mapDefault(metadata, "tags", ""), b.config.ForceTags, "udp")
+			service.ID = service.ID + ":udp"
+		} else {
+			service.Tags = combineTags(
+				mapDefault(metadata, "tags", ""), b.config.ForceTags)
+		}
+
+		id := mapDefault(metadata, "id", "")
+		if id != "" {
+			service.ID = id
+		}
 
-	id := mapDefault(metadata, "id", "")
-	if id != "" {
-		service.ID = id
+		service.Attrs = metadata
```

Comment:
This looks like a subtle difference, by assigning the `service.Attrs = metadata` here means that the metadata still contains the `id, tags, name` fields? Note that in the original code, there are three calls to `delete(metadata, ...)` to remove those fields before assigning `service.Attrs = metadata`. Is this any cause for concern or change in registration behavior?

Reply:
(I realize it's been almost two years since this PR was posted, and I doubt there will be any further activity on this project, but this PR solves a problem I was just researching in how to coordinate a service name transition period!)